libraries/rodeos/include/b1/rodeos/callbacks/filter.hpp

```diff
@@ -16,10 +16,28 @@ struct filter_callbacks {
       derived().get_filter_callback_state().push_data(data.data(), data.size());
    }
 
+   int32_t clock_gettime(int32_t id, uint64_t* data) {
+      std::chrono::nanoseconds result;
+      if (id == 0) { // CLOCK_REALTIME
+         result = std::chrono::system_clock::now().time_since_epoch();
```

Comment:
I thought we were avoiding non-determinism in rodeos.

Reply:
Also, libc `time` uses `current_time`.  The divergence between which functions return block time and which functions return the time at which the filter is running is likely to be confusing.

Reply:
This PR is to support running performance tests on portions of filter code.

Reply:
We goofed when we named the current block time `current_time` instead of `current_block_time`. We goofed again by hooking libc `time` to it. It'd be a shame to lose `std::chrono` for perf testing now because of our blunder back then.

Reply:
I don't really think that those were mistakes.  They make perfect sense from the POV of a contract, where block time is the only time.